Makefile

```diff
@@ -118,10 +118,18 @@ lint: lint-contracts-rs
 	$(CARGO) clippy --all-targets --all-features -- -D warnings -A renamed_and_removed_lints
 	cd smart_contracts/contract && $(CARGO) clippy --all-targets -- -D warnings -A renamed_and_removed_lints
 
-.PHONY: audit
-audit:
+.PHONY: audit-rs
+audit-rs:
 	$(CARGO) audit --ignore RUSTSEC-2020-0071 --ignore RUSTSEC-2020-0159
 
+.PHONY: audit-as
+audit-as:
+	@# Attempts to fix vulnerabilities and run a scan after which fails if there are still prod vulnerabilities with a moderate level or above.
+	cd smart_contracts/contract_as && $(NPM) audit fix --production && $(NPM) audit --production --audit-level=moderate
```

Comment:
I have checked and you are right, I have seen that the package step clones the repo one more time, so the changes would be lost.
Because of this I need to find a way to persist the change. 
I could as a quick solution apply the fix again during the packaging phase until a more elegan solution is found.

Reply:
something like this:

```suggestion
	cd smart_contracts/contract_as && $(NPM) audit --production --audit-level=moderate
```

Reply:
@TomVasile I think we can start with this as you proposed.

I was looking for ways to cache the changes and have them included in the package/release when doing npm audit fix as they are non-breaking.

Repeating the audit step adds too much redundancy even if the cost is relatively low compared to the median CI job run, and it does not carry the expected results (and it's not really following best practices).

In the current drone setup, to do it properly in that way, you should start caching the build results between steps, but it would be an even better way to achieve it if you have something automatically commit and push the easy automated npm audit fixes (Something like bors or GH actions can do it).

Reply:
@TomVasile I think we can start with this as you proposed.

I was looking for ways to cache the changes and have them included in the package/release when doing npm audit fix as they are non-breaking.

Repeating the audit step adds too much redundancy even if the cost is relatively low compared to the median CI job run, and it does not carry the expected results (and it's not really following best practices).

In the current drone setup, to do it properly in that way, you should start caching the build results between steps, but it would be an even better way to achieve it if you have something automatically commit and push the easy automated npm audit fixes (Something like bors or GH actions can do it).